modules/vulcan-testkit/src/main/scala/fs2/kafka/vulcan/testkit/SchemaSuite.scala

```diff
@@ -0,0 +1,76 @@
+package fs2.kafka.vulcan.testkit
+
+import fs2.kafka.vulcan.SchemaRegistryClientSettings
+import munit.FunSuite
+import vulcan.Codec
+import org.apache.avro.SchemaCompatibility
+import io.confluent.kafka.schemaregistry.avro.AvroSchema
+import cats.effect.IO
+import org.apache.avro.Schema
+
+trait CompatibilityChecker[F[_]] {
+  def checkReaderCompatibility[A](
+    reader: Codec[A],
+    writerSubject: String
+  ): F[SchemaCompatibility.SchemaPairCompatibility]
+
+  def checkWriterCompatibility[A](
+    writer: Codec[A],
+    readerSubject: String
+  ): F[SchemaCompatibility.SchemaPairCompatibility]
+}
+
+trait SchemaSuite extends FunSuite {
+
+  private def codecAsSchema[A](codec: Codec[A]) = codec.schema.fold(e => fail(e.message), ok => ok)
+
+  def compatibilityChecker(
+    clientSettings: SchemaRegistryClientSettings[IO],
+    name: String = "schema-compatibility-checker"
+  ) = new Fixture[CompatibilityChecker[IO]](name) {
+    private var checker: CompatibilityChecker[IO] = null
```

Comment:
Couldn't this be a lazy val rather than a var initialized in `beforeAll`?

Reply:
I guess that might mess with test timings then, as the first test to use the fixture would have the spin-up time bundled in?

Reply:
looking into the effect in question, don't think its actually doing any IO, so shouldn't add too much overhead to the first test to use it, that said I think I still find spinning up in the beforeAll slightly neater, even if the presence of `null` in the codebase makes me feel slightly ill...

Reply:
Yeah, it probably doesn't matter much either way.